src/Requests/PostEvent/JobSucceededWithWarningEventData.php

```diff
@@ -0,0 +1,43 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Keboola\NotificationClient\Requests\PostEvent;
+
+use Keboola\NotificationClient\Requests\EventDataInterface;
+
+class JobSucceededWithWarningEventData implements EventDataInterface
+{
+    private string $projectId;
+    private string $projectName;
+    private string $errorMessage;
+    private JobData $jobData;
+
+    public function __construct(string $projectId, string $projectName, string $errorMessage, JobData $jobData)
+    {
+        $this->projectId = $projectId;
+        $this->projectName = $projectName;
+        $this->errorMessage = $errorMessage;
+        $this->jobData = $jobData;
+    }
+
+    /**
+     * @return mixed
+     */
+    public function jsonSerialize()
```

Comment:
Nevraci to `mixed`, ale vzdycky `array`. A neslo by to dat rovnou jako return type? (nevim jesli to PHP sezere nebo ne, ale mohlo by)
```suggestion
    public function jsonSerialize(): array
```

Reply:
Okej, to som skopcil z JobFailed :)

Reply:
Jo to je kvoli tomu, ze to implementuje JsonSerializable interface:

![Screenshot 2021-08-30 at 13 19 31](https://user-images.githubusercontent.com/1488015/131331799-22183cc7-b176-4e6b-aa47-132994d9856e.png)

Reply:
No jasne, protoze obecne to klidne muze vracet i scalar (string, cislo i boolean sam o sobe je validni json). Ale ty vracis vzdycky array, tak si myslim, ze se tam ten return type hodi vic.

Reply:
Ale pak nemozes implementovat ten interface, to musi byt stejne

Reply:
A nebo ne? 🤔  skusim

Reply:
Mě to taky nějak zlobilo s tím array (ale možná to bylo taky starší PHP), tak jsem tam buď dával mixed nebo @inheritdoc

Reply:
Teraz to vyzera, ze to prejde, ale tiez som tomu neveril :)